### PR TITLE
push layers in order

### PR DESCRIPTION
**Pull request by sowu880**

### CODE REVIEW

File: src/Microsoft.Health.Fhir.TemplateManagement/OCIFileManager.cs [hunk 1]
```diff
@@ -72,7 +72,10 @@ public void PackOCIImage(bool ignoreBaseLayers = false)
 
         public async Task PushOCIImageAsync()
         {
-            await _orasClient.PushImageAsync(_overlayFS.WorkingImageLayerFolder);
+            var rawLayers = _overlayFS.ReadImageLayers();
+            var fileLayers = _overlayOperator.ExtractOCIFileLayers(rawLayers);
+            var sortedLayers = _overlayOperator.SortOCIFileLayersBySequenceNumber(fileLayers);
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement/OCIFileManager.cs:77]: Seems we can always extract file layers in sequence number order. Can we merge ExtractOCIFileLayers and SortOCIFileLayersBySequenceNumber into one function?

File: src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs [hunk 2]
```diff
@@ -30,18 +31,21 @@ public async Task PullImageAsync(string outputFolder)
             await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
         }
 
-        public async Task PushImageAsync(string inputFolder)
+        public async Task PushImageAsync(string inputFolder, List<string> filePathList)
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs:34]: Can we use a single argument instead of parsing folder and filenameList?

Comment (sowu880) [reply src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs:34]: Seems not easy to separate working folder and file name which combined in parameter.

File: src/Microsoft.Health.Fhir.TemplateManagement/OCIFileManager.cs (hunk 1 above)
Comment (sowu880) [reply src/Microsoft.Health.Fhir.TemplateManagement/OCIFileManager.cs:77]: SortBySequenceNumber function will be removed if using manifest next release version.

File: src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs (hunk 2 above)
Comment (qiwjin) [reply src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs:34]: it's a hot-fix PR. we can discuss it later.

**Review by qiwjin — APPROVED** [review]

**Review by moria97 — APPROVED** [review]

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**